### PR TITLE
Add kernel RaggedRange for CPU and WebGL backend

### PR DESCRIPTION
**Pull request by ahmedsabie**
To see the logs from the Cloud Build CI, please join either our [discussion](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs) or [announcement](https://groups.google.com/a/tensorflow.org/forum/#!forum/tfjs-announce) mailing list.

- - -
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6871)

### CODE REVIEW

File: tfjs-backend-cpu/src/kernels/RaggedRange.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC. All Rights Reserved.
```
Comment (mattsoulanille) [inline tfjs-backend-cpu/src/kernels/RaggedRange.ts:3]:
```suggestion
 * Copyright 2022 Google LLC.
```

File: tfjs-backend-cpu/src/kernels/RaggedRange_impl.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC. All Rights Reserved.
```
Comment (mattsoulanille) [inline tfjs-backend-cpu/src/kernels/RaggedRange_impl.ts:3]:
```suggestion
 * Copyright 2022 Google LLC.
```

File: tfjs-backend-webgl/src/kernels/RaggedRange.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC. All Rights Reserved.
```
Comment (mattsoulanille) [inline tfjs-backend-webgl/src/kernels/RaggedRange.ts:3]:
```suggestion
 * Copyright 2022 Google LLC.
```

File: tfjs-core/src/ops/ragged_range.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC. All Rights Reserved.
```
Comment (mattsoulanille) [inline tfjs-core/src/ops/ragged_range.ts:3]:
```suggestion
 * Copyright 2022 Google LLC.
```

File: tfjs-core/src/ops/ragged_range_test.ts [hunk 5]
```diff
@@ -0,0 +1,190 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC. All Rights Reserved.
```
Comment (mattsoulanille) [inline tfjs-core/src/ops/ragged_range_test.ts:3]:
```suggestion
 * Copyright 2022 Google LLC.
```

**Review by mattsoulanille — CHANGES_REQUESTED** [review]
Just a quick review removing "all rights reserved" from the license headers of new files (don't change existing files).

**Review by lina128 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6871)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @ahmedsabie and @pyu10055)

___
*[`tfjs-backend-cpu/src/kernels/RaggedRange_impl.ts` line 20 at r1](https://reviewable.io/reviews/tensorflow/tfjs/6871#-ND4vvLzBewnOj1ILAUz:-ND4vvLzBewnOj1ILAV-:b-mzonpm) ([raw file](https://github.com/tensorflow/tfjs/blob/a515495cf6009d4c88b583c318647923a4cf0a1b/tfjs-backend-cpu/src/kernels/RaggedRange_impl.ts#L20)):*
> ```XML
> import {DataType, TypedArray, util} from '@tensorflow/tfjs-core';
> 
> const INT32_MAX = 2147483647;
> ```

Why not use Number.MAX_SAFE_INTEGER?



_Code quote:_
```XML
const INT32_MAX = 2147483647
```

**Review by pyu10055 — APPROVED** [review]
thank you for adding this op.

> Reviewed 12 of 12 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/6871#-:-NDFh3Jh3kU0VfVjkc0K:b5jbvdj)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @ahmedsabie)

**Review by mattsoulanille — APPROVED** [review]
LGTM

**Pull request merged**